File: library/src/main/java/me/proxer/library/entity/list/Tag.java
package me.proxer.library.entity.list;

import com.squareup.moshi.Json;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Value;
import me.proxer.library.entity.ProxerIdItem;
import me.proxer.library.enums.TagSubType;
import me.proxer.library.enums.TagType;

import javax.annotation.Nullable;

/**
 * Entity representing a single tag.
 *
 * @author Ruben Gees
 */
@Value
@EqualsAndHashCode(onParam = @__({@Nullable}))
public class Tag implements ProxerIdItem {

    /**
     * Returns the id.
     */
    @Getter(onMethod = @__({@Override}))
    @Json(name = "id")
    private String id;

    /**
     * Returns the type.
     */
    @Json(name = "type")
    private TagType type;

    /**
     * Returns the name.
     */
    @Json(name = "tag")
    private String name;

    /**
     * Returns the name.
     */
    @Json(name = "description")
    private String description;

    /**
     * Returns the type.
     */
    @Json(name = "subtype")
    private TagSubType subType;

    /**
     * Returns the type.
     */
    @Json(name = "is_spoiler")
    private boolean isSpoiler;
}

File: library/src/main/kotlin/me/proxer/library/entity/list/Tag.kt
package me.proxer.library.entity.list

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import me.proxer.library.api.NumberBasedBoolean
import me.proxer.library.entity.ProxerIdItem
import me.proxer.library.enums.TagSubType
import me.proxer.library.enums.TagType

/**
 * Entity representing a single tag.
 *
 * @property type The type.
 * @property name The name.
 * @property description The description.
 * @property subType The sub type.
 * @property isSpoiler If this tag is a spoiler.
 *
 * @author Ruben Gees
 */
@JsonClass(generateAdapter = true)
data class Tag(
    @Json(name = "id") override val id: String,
    @Json(name = "type") val type: TagType,
    @Json(name = "tag") val name: String,
    @Json(name = "description") val description: String,
    @Json(name = "subtype") val subType: TagSubType,
    @field:NumberBasedBoolean @Json(name = "is_spoiler") val isSpoiler: Boolean
) : ProxerIdItem
